OpenAI-DotNet/Chat/Function.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Text.Json.Nodes;
+using System.Text.Json.Serialization;
+
+namespace OpenAI.Chat
+{
+    public sealed class Function
+    {
+        public Function(string name, string description = null, JsonObject parameters = null)
+        {
+            if(string.IsNullOrWhiteSpace(name))
+                throw new ArgumentException("Function name cannot be null or whitespace.", nameof(name));
+
+            if(name.Length > 64)
+                throw new ArgumentException("Function name cannot be longer than 64 characters.", nameof(name));
+
+            if (!char.IsLetter(name[0]) && name[0] != '_')
+                throw new ArgumentException("Function name must begin with a letter or underscore.", nameof(name));
```

Comment:
```suggestion
            if (!char.IsLetter(name[0]) && name[0] != '_')
            {
                throw new ArgumentException("Function name must begin with a letter or underscore.", nameof(name));
            }
```

Reply:
I decided that this might be a bad idea for someone somewhere sometime.  I commented this one.